src/python/pants/backend/python/rules/coverage.py

```diff
@@ -158,12 +178,9 @@ class PytestCoverageDataCollection(CoverageDataCollection):
     element_type = PytestCoverageData
 
 
-# TODO(#7490): Once we have support for `await Get` of singletons, we can remove this class. We only
-# have it so that we can avoid `pytest_runner.py` from awaiting `CoverageConfig` if coverage is not
-# being used.
 @dataclass(frozen=True)
 class CoverageConfigRequest:
-    pass
+    targets: FrozenOrderedSet[Target]
```

Comment:
This change results in better caching. Previously, if a completely unrelated target changed, it would result in `pytest_runner.py` being invalidated because the CoverageConfig would have changed.